hearth/lib/hearth/middleware/send.rb

```diff
@@ -71,25 +71,69 @@ def call(input, context)
       private
 
       def stub_response(input, context, output)
-        stub = @stubs.next(context.operation_name)
-        log_debug(context, "Stubbing response with stub: #{stub}")
-        apply_stub(stub, input, context, output)
-        log_debug(context, "Stubbed response: #{context.response.inspect}")
-        return unless context.response.body.respond_to?(:rewind)
-
-        context.response.body.rewind
+        context.tracer.in_span(
+          'Middleware.Stub',
+          attributes: request_attributes(context)
+        ) do |span|
+          stub = @stubs.next(context.operation_name)
+          log_debug(context, "Stubbing response with stub: #{stub}")
+          apply_stub(stub, input, context, output)
+          log_debug(context, "Stubbed response: #{context.response.inspect}")
+          response_attributes(span, context)
+        end
+        rewind_response_body(context)
       end
 
       def send_request(context, output)
-        log_debug(context, "Sending request: #{context.request.inspect}")
-        @client.transmit(
-          request: context.request,
-          response: context.response,
-          logger: context.config.logger
-        )
-        log_debug(context, "Received response: #{context.response.inspect}")
-      rescue Hearth::NetworkingError => e
-        output.error = e
+        context.tracer.in_span(
```

Comment:
It makes sense to maybe move these blocks outside of send_request and stub_response for readability, or create a wrapper method.